Windows CI errors:
```
FAIL: //py_test_dir/tensorflow/python/util:tf_stack_test (see T:/tmp/bigvaudl/execroot/org_tensorflow/bazel-out/x64_windows-opt/testlogs/py_test_dir/tensorflow/python/util/tf_stack_test/test_attempts/attempt_1.log)
FAIL: //py_test_dir/tensorflow/python/util:tf_stack_test (see T:/tmp/bigvaudl/execroot/org_tensorflow/bazel-out/x64_windows-opt/testlogs/py_test_dir/tensorflow/python/util/tf_stack_test/test_attempts/attempt_2.log)
FAIL: //py_test_dir/tensorflow/python/util:tf_stack_test (see T:/tmp/bigvaudl/execroot/org_tensorflow/bazel-out/x64_windows-opt/testlogs/py_test_dir/tensorflow/python/util/tf_stack_test/test.log)

FAILED: //py_test_dir/tensorflow/python/util:tf_stack_test (Summary)
      T:/tmp/bigvaudl/execroot/org_tensorflow/bazel-out/x64_windows-opt/testlogs/py_test_dir/tensorflow/python/util/tf_stack_test/test.log
      T:/tmp/bigvaudl/execroot/org_tensorflow/bazel-out/x64_windows-opt/testlogs/py_test_dir/tensorflow/python/util/tf_stack_test/test_attempts/attempt_1.log
      T:/tmp/bigvaudl/execroot/org_tensorflow/bazel-out/x64_windows-opt/testlogs/py_test_dir/tensorflow/python/util/tf_stack_test/test_attempts/attempt_2.log
INFO: From Testing //py_test_dir/tensorflow/python/util:tf_stack_test:
==================== Test output for //py_test_dir/tensorflow/python/util:tf_stack_test:
Running tests under Python 3.7.4: C:\Python37\python.exe
[ RUN      ] TFStackTest.testFormatStackSelfConsistency
INFO:tensorflow:time(__main__.TFStackTest.testFormatStackSelfConsistency): 0.01s
I0115 17:41:43.657681 20736 test_util.py:2071] time(__main__.TFStackTest.testFormatStackSelfConsistency): 0.01s
[       OK ] TFStackTest.testFormatStackSelfConsistency
[ RUN      ] TFStackTest.testFrameSummaryEquality
INFO:tensorflow:time(__main__.TFStackTest.testFrameSummaryEquality): 0.0s
I0115 17:41:43.657681 20736 test_util.py:2071] time(__main__.TFStackTest.testFrameSummaryEquality): 0.0s
[       OK ] TFStackTest.testFrameSummaryEquality
[ RUN      ] TFStackTest.testFrameSummaryEqualityAndHash
INFO:tensorflow:time(__main__.TFStackTest.testFrameSummaryEqualityAndHash): 0.0s
I0115 17:41:43.659635 20736 test_util.py:2071] time(__main__.TFStackTest.testFrameSummaryEqualityAndHash): 0.0s
[       OK ] TFStackTest.testFrameSummaryEqualityAndHash
[ RUN      ] TFStackTest.testLastUserFrame
INFO:tensorflow:time(__main__.TFStackTest.testLastUserFrame): 0.0s
I0115 17:41:43.660615 20736 test_util.py:2071] time(__main__.TFStackTest.testLastUserFrame): 0.0s
[  FAILED  ] TFStackTest.testLastUserFrame
[ RUN      ] TFStackTest.test_session
[  SKIPPED ] TFStackTest.test_session
======================================================================
FAIL: testLastUserFrame (__main__.TFStackTest)
TFStackTest.testLastUserFrame
----------------------------------------------------------------------
Traceback (most recent call last):
  File "\\?\T:\tmp\Bazel.runfiles_s9zl8mw2\runfiles\org_tensorflow\py_test_dir\tensorflow\python\util\tf_stack_test.py", line 57, in testLastUserFrame
    self.assertRegex(frame.line, "# COMMENT")
AssertionError: Regex didn't match: '# COMMENT' not found in '_source_filter_stacks[thread_key][-1].internal_set)'
```

This is unrelated to my CL. Looking at the code, it seems that commit 73a6839fb3b61bd307630b612261292bc562f527 touched this area recently. It actually dropped windows file patterns from consideration.
```"tensorflow\\python"``` is no longer considered as a pattern for IsInternalFrameForFilename. It is unclear to me whether this change is correct.
@edloper: Can you please assist ?

--- a/tensorflow/python/keras/utils/BUILD
+++ b/tensorflow/python/keras/utils/BUILD
@@ -449,3 +449,23 @@ tf_py_test(
         "@absl_py//absl/testing:parameterized",
     ],
 )
+
+tf_py_test(
+    name = "losses_utils_test",
+    size = "small",
+    srcs = ["losses_utils_test.py"],
+    python_version = "PY3",
+    deps = [
+        "//tensorflow/python:constant_op",
+        "//tensorflow/python:framework_ops",
+        "//tensorflow/python:framework_test_lib",
+        "//tensorflow/python:ops",
+        "//tensorflow/python:platform_test",
+        "//tensorflow/python/eager:context",
+        "//tensorflow/python/keras",
+        "//tensorflow/python/keras:combinations",
+        "//tensorflow/python/ops/ragged:ragged_array_ops",
+        "//tensorflow/python/ops/ragged:ragged_concat_ops",
+        "//tensorflow/python/ops/ragged:ragged_factory_ops",
+    ],
+)

--- a/tensorflow/python/keras/utils/losses_utils_test.py
+++ b/tensorflow/python/keras/utils/losses_utils_test.py
@@ -1,0 +1,85 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for losses_utils."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.framework import constant_op
+from tensorflow.python.framework import test_util
+from tensorflow.python.framework import ops
+from tensorflow.python.keras import combinations
+from tensorflow.python.keras.utils import losses_utils
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops.ragged import ragged_factory_ops
+from tensorflow.python.platform import test
+
+
+@combinations.generate(combinations.combine(mode=['graph', 'eager']))
+class RemoveSqueezableTest(test_util.TensorFlowTestCase):
+  """Test remove_squeezable_dimensions
+  """
+  def test_ragged_3d_same_shape(self):
+    """ shape (2, (sequence={1, 2}), 3)
+    """
+    x = ragged_factory_ops.constant([[[1, 2, 3]], [[4, 5, 6], [7, 8, 9]]])
+    rank = x.shape.ndims
+    x_p, _ = losses_utils.remove_squeezable_dimensions(x, x)
+    self.assertEqual(x_p.shape.ndims, rank)
+
+  def test_ragged_3d_4d_squeezable(self):
+    """ shapes:
+        x: (2, (sequence={1, 2}), 3)
+        y: (2, (sequence={1, 2}), 3, 1)
+    """
+    x = ragged_factory_ops.constant([[[1, 2, 3]], [[4, 5, 6], [7, 8, 9]]])
+    y = array_ops.expand_dims(x, axis=-1)
+    self.assertEqual(x.shape.ndims, 3)
+    self.assertEqual(y.shape.ndims, 4)
+    _, y_p = losses_utils.remove_squeezable_dimensions(x, y)
+    y_p.shape.assert_is_compatible_with(x.shape)
+    self.assertEqual(y_p.shape.ndims, 3)
+
+    x_p, _ = losses_utils.remove_squeezable_dimensions(y, x)
+    x_p.shape.assert_is_compatible_with(x.shape)
+    self.assertEqual(x_p.shape.ndims, 3)
+
+  def test_dense_2d_3d_squeezable(self):
+    x = constant_op.constant([[1, 2], [3, 4]])
+    y = constant_op.constant([[[1], [2]], [[3], [4]]])
+    _, y_p = losses_utils.remove_squeezable_dimensions(x, y)
+    y_p.shape.assert_is_compatible_with(x.shape)
+    self.assertEqual(y_p.shape.ndims, x.shape.ndims)
+    x_p, _ = losses_utils.remove_squeezable_dimensions(y, x)
+    x_p.shape.assert_is_compatible_with(x.shape)
+
+class RemoveSqueezableTestGraphOnly(test_util.TensorFlowTestCase):
+  """Test remove_squeezable_dimensions (graph-mode only).
+  """
+  def test_placeholder(self):
+    """Test dynamic rank tensors.
+    """
+    with ops.Graph().as_default():
+      x = array_ops.placeholder_with_default([1., 2., 3.], shape=None)
+      y = array_ops.placeholder_with_default([[1.], [2.], [3.]], shape=None)
+      _, y_p = losses_utils.remove_squeezable_dimensions(x, y)
+      y_p.shape.assert_is_compatible_with(x.shape)
+      self.assertAllEqual(array_ops.shape(x), array_ops.shape(y_p))
+      x_p, _ = losses_utils.remove_squeezable_dimensions(y, x)
+      x_p.shape.assert_is_compatible_with(x.shape)
+
+if __name__ == '__main__':
+  test.main()

